src/utils/detect-server-settings.mjs

```diff
@@ -182,6 +186,71 @@ const getSettingsFromFramework = (framework) => {
 
 const hasDevCommand = (framework) => Array.isArray(framework.dev.commands) && framework.dev.commands.length !== 0
 
+/**
+ * The new build setting detection with build systems and frameworks combined
+ * @param {string} projectDir
+ */
+const detectSettings = async (projectDir) => {
+  const fs = new NodeFS()
+  const project = new Project(fs, projectDir)
+
+  return await project.getBuildSettings()
+}
+
+/**
+ *
+ * @param {import('./types.js').BaseServerSettings | undefined} frameworkSettings
+ * @param {import('@netlify/build-info').Settings[]} newSettings
+ * @param {Record<string, Record<string, any>>} [metadata]
+ */
+const detectChangesInNewSettings = (frameworkSettings, newSettings, metadata) => {
```

Comment:
What do you think of mapping through they keys of `frameworkSettings` and check if their values differ with those of `newSettings` instead of manual `ifs`?

Reply:
not possible as keys are named differently:

```
if (frameworkSettings?.command !== setting?.devCommand) {
```

Reply:
Since its only one key and the other keys names are consistent between frameworkSettings & settings does it make sense to rename the key then map? ie `frameworkSettings.devCommand = frameworkSettings.command`

Reply:
that's not even worth the effort (this is JUST internal reporting for me for some days then it will be completely removed)